metashare/api/models.py

```diff
@@ -458,6 +496,13 @@ class Task(
     def __str__(self):
         return self.name
 
+    def save(self, *args, **kwargs):
+        ret = super().save(*args, **kwargs)
+        # To update the project's status:
+        if self.project.should_update_status():
+            self.project.save()
+        return ret
```

Comment:
@wlonk Given this change, I think we'll want some cleanup to the `Task.finalize_status_completed` method so that it doesn't require multiple saves.